src/internal/components/autosuggest-input/index.tsx

```diff
@@ -131,8 +132,10 @@ const AutosuggestInput = React.forwardRef(
       ) {
         return;
       }
-      closeDropdown();
-      fireNonCancelableEvent(onBlur, null);
+      if (!preventCloseOnBlurRef.current) {
```

Comment:
Some of this focus handling (not 100% sure it's this bit or elsewhere) seems to create a tab trap: if opening the `autosuggest-input` page and then tabbing through, it gets stuck cycling between the input and the button in the dropdown

Reply:
The tab trap is intended. It is controlled by the `trapFocus` property of the dropdown.

Reply:
OK yes, I see now that it didn't actually change. Still not quite sure I understand _why_ we'd ever want a tab trap here, but I guess that's outside the scope of this PR.